internal/resources/cluster/tkgaws/resource_tkg_aws.go

```diff
@@ -0,0 +1,967 @@
+	flattenClusterNetwork := make(map[string]interface{})
+
+	if cluster == nil {
+		return nil
+	}
+
+	flattenClusterNetwork[apiServerPortKey] = cluster.APIServerPort
+
+	pds := make([]interface{}, 0)
+
+	for _, pd := range cluster.Pods {
+		pds = append(pds, flattenTKGAWSNetworkRange(pd))
+	}
+
+	flattenClusterNetwork[podsKey] = pds
+
+	svs := make([]interface{}, 0)
+
+	for _, sv := range cluster.Services {
+		svs = append(svs, flattenTKGAWSNetworkRange(sv))
+	}
+
+	flattenClusterNetwork[servicesKey] = svs
+
+	return []interface{}{flattenClusterNetwork}
+}
+
+func flattenTKGAWSNetworkRange(ranges *tkgawsmodel.VmwareTanzuManageV1alpha1ClusterInfrastructureTkgawsNetworkRange) (data interface{}) {
+	flattenRanges := make(map[string]interface{})
+
+	if ranges == nil {
+		return nil
+	}
+
+	flattenRanges[cidrBlocksKey] = ranges.CidrBlocks
+
+	return flattenRanges
+}
+
+var tkgAWSProvider = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "Provider Network specifies provider specific network information for the cluster",
+	Required:    true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			subnetsKey: tkgAWSSubnets,
+			vpcKey:     tkgAWSVPC,
+		},
+	},
+}
+
+func expandTKGAWSProviderNetwork(data []interface{}) (provider *tkgawsmodel.VmwareTanzuManageV1alpha1ClusterInfrastructureTkgawsProviderNetwork) {
+	if len(data) == 0 || data[0] == nil {
+		return provider
+	}
+
+	lookUpProvider, _ := data[0].(map[string]interface{})
+	provider = &tkgawsmodel.VmwareTanzuManageV1alpha1ClusterInfrastructureTkgawsProviderNetwork{
+		Subnets: []*tkgawsmodel.VmwareTanzuManageV1alpha1ClusterInfrastructureTkgawsSubnet{},
+		Vpc:     &tkgawsmodel.VmwareTanzuManageV1alpha1ClusterInfrastructureTkgawsVPC{},
+	}
+
+	if v, ok := lookUpProvider[subnetsKey]; ok {
+		subnets, _ := v.([]interface{})
+		for _, sn := range subnets {
+			provider.Subnets = append(provider.Subnets, expandTKGAWSSubnets(sn))
+		}
+	}
+
+	if v, ok := lookUpProvider[vpcKey]; ok {
+		if v1, ok := v.([]interface{}); ok {
+			provider.Vpc = expandTKGAWSVPC(v1)
+		}
+	}
+
+	return provider
+}
+
+func flattenTKGAWSProviderNetwork(provider *tkgawsmodel.VmwareTanzuManageV1alpha1ClusterInfrastructureTkgawsProviderNetwork) (data []interface{}) {
+	flattenProviderNetwork := make(map[string]interface{})
+
+	if provider == nil {
+		return nil
+	}
+
+	sns := make([]interface{}, 0)
+
+	for _, sn := range provider.Subnets {
+		sns = append(sns, flattenTKGAWSSubnets(sn))
+	}
+
+	flattenProviderNetwork[subnetsKey] = sns
+
+	if provider.Vpc != nil {
+		flattenProviderNetwork[vpcKey] = flattenTKGAWSVPC(provider.Vpc)
+	}
+
+	return []interface{}{flattenProviderNetwork}
+}
+
+var tkgAWSSubnets = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "Optional list of subnets used to place the nodes in the cluster",
+	Optional:    true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
```

Comment:
Please add computed to the fields which gets autofilled. We also have certain fields like vpc_id and subnet id right, please handle those fields as well

Reply:
They have already been taken care.